src/MSBuild/XMake.cs

```diff
@@ -296,6 +298,9 @@ string[] args
                 DumpCounters(false /* log to console */);
             }
 
+            // Send OpenTelemetry before exiting
+            OpenTelemetryManager.Instance.Shutdown();
```

Comment:
We don't seem to call this for the API usage case.
Let's try to see if we can have better place to control lifetime of the telemetry manager (probably BuildManager?)

Reply:
Oh that's an oversight given we don't have the collector in VS yet. Now that I look at it, it makes more sense to do everything in EndBuild.

The point is that somewhere the collector has to be shut down, whereas the Exporter is ok to just flush (if collector is somewhere else).
Is it also a goal to send telemetry when people are using MSBuild via API on their own?